pkg/agent/storage/backcompat.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"crypto/x509"
+	"path/filepath"
+)
+
+func Backcompat(dir string) (Storage, error) {
+	j, err := JSONFile(filepath.Join(dir, "data.json"))
```

Comment:
Hmm, I need to do some more changes here related to upgrade/downgrade. If the file doesn't exist, we should prepopulate it from the legacy storage. If there is an upgrade/downgrade/upgrade, we need to figure out what to do.... check mtimes to figure out which one to take?

Reply:
I think that checking the modified time is what makes more sense, and take the latest one.

Reply:
Ok, I'll see what I can do.